ui/src/app/settings/reducers/general/general.js

```diff
@@ -0,0 +1,25 @@
+import produce from "immer";
+
+const general = produce(
+  (draft, action) => {
+    switch (action.type) {
+      case "FETCH_GENERAL_OSINFO_START":
+        draft.loading = true;
+        break;
+      case "FETCH_GENERAL_OSINFO_SUCCESS":
+        draft.loading = false;
+        draft.loaded = true;
+        draft.items = action.payload;
+        break;
+      default:
+        return draft;
+    }
+  },
+  {
+    loading: false,
+    loaded: false,
+    items: {}
```

Comment:
Hmm admittedly this is a bit of a weird one. I feel like `items` should be used for arrays of the same type of entity, and maybe this should just have the different data objects at the top level e.g:

``` js
{
  loading: false,
  loaded: false,
  osInfo: null,
  hweKernels: null,
  minHweKernels: null,
  ...
}
```

That way when accessing them you would access `state.general.osInfo` instead of `state.general.items.osInfo` which seems unnecessarily nested.

Also I'm not sure if it's a good or bad idea to set the default to whatever data structure we expect to receive the data, so I've opted for `null` here - maybe @squidsoup or @huwshimi have a better idea for this.

Reply:
Done